ton/wallet/address.go

```diff
@@ -29,10 +30,37 @@ func AddressFromPubKey(key ed25519.PublicKey, ver Version, subwallet uint32) (*a
 	return addr, nil
 }
 
-func GetStateInit(pubKey ed25519.PublicKey, ver Version, subWallet uint32) (*tlb.StateInit, error) {
-	code, err := getCode(ver)
+func GetWalletVersion(ctx context.Context, api TonAPI, addr *address.Address) (Version, error) {
+	master, err := api.CurrentMasterchainInfo(ctx)
+	if err != nil {
+		return 0, err
+	}
+
+	account, err := api.GetAccount(ctx, master, addr)
```

Comment:
Lets accept *tlb.Account as argument instead of address, it will make method more flexible.

And instead of error maybe return something like Unknown version